pkg/controller/elasticsearch/remotecluster/elasticsearch.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package remotecluster
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/license"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/tracing"
+	esclient "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/services"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"go.elastic.co/apm"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("remotecluster")
+
+// UpdateRemoteCluster updates the remote clusters in the persistent settings by calling the Elasticsearch API.
+func UpdateRemoteCluster(
+	ctx context.Context,
+	c k8s.Client,
+	esClient esclient.Client,
+	licenseChecker license.Checker,
+	es esv1.Elasticsearch,
+) error {
+	span, _ := apm.StartSpan(ctx, "update_remote_clusters", tracing.SpanTypeApp)
+	defer span.End()
+
+	enabled, err := licenseChecker.EnterpriseFeaturesEnabled()
+	if err != nil {
+		return err
+	}
+	if !enabled {
```

Comment:
I didn't do that on purpose because I don't know what would be the technical consequences for the follower indices if it's only a temporary situation (i.e.  license expired but may be renewed shortly).